samcli/lib/build/app_builder.py

```diff
@@ -663,3 +706,89 @@ def _parse_builder_response(stdout_data: str, image_name: str) -> Dict:
             raise ValueError(msg)
 
         return cast(Dict, response)
+
+    @staticmethod
+    def _make_env_vars(function: Function, file_env_vars: Dict, inline_env_vars: Optional[Dict]) -> Dict:
+        """Returns the environment variables configuration for this function
+
+        Parameters
+        ----------
+        function : samcli.lib.providers.provider.Function
+            Lambda function to generate the configuration for
+
+        Returns
+        -------
+        dictionary
+            Environment variable configuration for this function
+
+        Raises
+        ------
+        samcli.commands.local.lib.exceptions.OverridesNotWellDefinedError
+            If the environment dict is in the wrong format to process environment vars
+
+        """
+
+        name = function.name
+        result = {}
+
+        if file_env_vars and inline_env_vars:
+            for file_env_var in file_env_vars.values():
+                if not isinstance(file_env_var, dict):
+                    reason = """
+                                Environment variables in incorrect format
+                                """
+                    LOG.debug(reason)
+                    raise OverridesNotWellDefinedError(reason)
+
+            for inline_env_var in inline_env_vars.values():
+                if not isinstance(inline_env_var, dict):
+                    reason = """
+                                Environment variables in incorrect format
```

Comment:
Maybe we can be more helpful here. like `Environment variables {inline_env_var} in incorrect format`

Reply:
makes sense